feat(Android): Add docs for Jetpack Compose performance metrics (#6510)
getsentry/sentry-docs: src/platforms/android/configuration/integrations/jetpack-compose.mdx

  - mobile
---

The `sentry-compose-android` library provides [Jetpack Compose](https://developer.android.com/jetpack/androidx/releases/compose) support in the following areas:
* `@Composable` performance metrics

> * Transaction and breadcrumb support for User interactions (clicks, swipes and scroll gestures)
> * Transaction and breadcrumb support for Navigation events

Reviewer: ```suggestion
* Transaction and breadcrumb support for user interactions (clicks, swipes and scroll gestures)
* Transaction and breadcrumb support for navigation events
```

> ## Performance Metrics

Reviewer 2: I'm thinking we should create a subpage per section, otherwise it's already a lot to read through. But no need to do it in this PR I guess, can be done later.

_(New in version 6.16.0)_

With Jetpack Compose performance metrics the Sentry Android SDK can automatically measure the initial `composition` and `rendering` time as performance spans of your `@Composable` UI elements.

<Note>

> This feature relies on Automatic Activity Instrumentation, check the <PlatformLink to="/performance/instrumentation/automatic-instrumentation/">performance documentation</PlatformLink> on how to get started first.

Reviewer 2: but should also work with Navigation, right? Should we mention it here as well?

> This feature relies on Automatic Activity Instrumentation, check the <PlatformLink to="/performance/instrumentation/automatic-instrumentation/">performance documentation</PlatformLink> on how to get started first.

Reviewer: ```suggestion
This feature relies on automatic Activity instrumentation. Read the <PlatformLink to="/performance/instrumentation/automatic-instrumentation/">performance documentation</PlatformLink> to learn how to get started.
```

</Note>

Once enabled you'll see two spans inside your transactions:
* `ui.compose.composition`, describing the time it took to execute your `@Composable`
* `ui.compose.rendering`, describing the time it took to render your `@Composable` onto the canvas

![Jetpack Compose Performance Metrics](jetpack-compose-performance-metrics.png)

> In order to use it, you need to decorate your `@Composable` body with the `SentryTraced` function.

Reviewer 2: should we move it under `Configuration` sub-section, similar to others here? It should go after `Installation` probably, to align with the rest of the page.

> Once enabled you'll see two spans inside your transactions:
> * `ui.compose.composition`, describing the time it took to execute your `@Composable`
> * `ui.compose.rendering`, describing the time it took to render your `@Composable` onto the canvas
> 
> ![Jetpack Compose Performance Metrics](jetpack-compose-performance-metrics.png)
> 
> In order to use it, you need to decorate your `@Composable` body with the `SentryTraced` function.

Reviewer: ```suggestion
Once performance metrics have been enabled, you'll see two spans inside your transactions:
* `ui.compose.composition`, describing the time it took to execute your `@Composable`
* `ui.compose.rendering`, describing the time it took to render your `@Composable` to the canvas

![Jetpack Compose Performance Metrics](jetpack-compose-performance-metrics.png)

In order to use it, you need to decorate your `@Composable` body with the `SentryTraced` function.
```
Is "decorate" the right word here? Haven't seen it used in this context before :)

Author: Good point, I think _wrap_ is a more common term, let's use it instead.

```kotlin
import io.sentry.compose.SentryTraced

@Composable
fun LoginScreen() {
  SentryTraced("login_screen") {
    Column {
      // ...
      Button(
          onClick = { TODO() }) {
          Text(text = "Login")
      }
    }
  }
}
```

> The `SentryTraced` function also provides a `enableUserInteractionTracing` argument, which automatically applies a `Modifier.testTag()`, thus automatically enabling the _User Interactions_ feature mentioned below as well.

Reviewer: ```suggestion
The `SentryTraced` function also provides an `enableUserInteractionTracing` argument, which automatically applies a `Modifier.testTag()`and enables the _User Interactions_ feature mentioned below.
```

### Installation

To add the Jetpack Compose integration, install the [Android SDK](/platforms/android/), then add the `sentry-compose-android` dependency using Gradle:

```groovy
implementation 'io.sentry:sentry-android:{{ packages.version('sentry.java.android', '6.2.0') }}'
implementation 'io.sentry:sentry-compose-android:{{ packages.version('sentry.java.compose', '6.2.0') }}'
```

## User Interactions
_(New in version 6.10.0)_

> The Sentry User Interactions feature for Jetpack Compose can detect click, scroll and swipe gestures. For every gesture the SDK can automatically collects breadcrumbs and launch transactions.

Reviewer 2: ```suggestion
The Sentry User Interactions feature for Jetpack Compose can detect click, scroll and swipe gestures. For every gesture the SDK can automatically collect breadcrumbs and launch transactions.
```
